### ISSUE TITLE
Respect property to disable targeting pack caching for runtime build

### ISSUE THREAD

**Issue by rainersigwald** [dotnet/sdk#22834]
#19570 introduced caching in `ResolveTargetingPackAssets`. I'm not aware of complaints from "normal" users, but for the `dotnet/runtime` repo, which _produces_ those targeting packs _during its build_, the caching can preserve stale data.

> I think this is becoming an issue in between using `dotnet build` on source projects where the caching feature is not disabled as the env var is only set on the build scripts and then using `dotnet build` or `dotnet test` on a test project where the platform manifest is needed.
> 
> @rainersigwald is it possible that we can also support an MSBuild property to disable the targeting pack caching?

_Originally posted by @safern in https://github.com/dotnet/runtime/issues/61451#issuecomment-982812791_

**Comment by rainersigwald** [comment]
> Yeah, we can add an input to the task and pass a property through to it. It would require an SDK update. @marcpopMSFT would you have any objection to adding such a feature to an SDK task when the only expected consumer is the runtime build? I personally don't . . .

_Originally posted by @rainersigwald in https://github.com/dotnet/runtime/issues/61451#issuecomment-982815610_

**Comment by rainersigwald** [comment]
> captured the "change" that happens in dotnet/runtime that causes it to break

Yeah, is this just "files magically appear in those directories" or is it more complicated?

**Comment by ericstj** [comment]
I'm not 100% certain as I haven't looked at a repro personally.  I can imagine things like file existence and modified time might be interesting inputs to capture in the cache key, similar to target inputs/outputs.

**Comment by rainersigwald** [comment]
We opted not to have existence/timestamp checks for cache invalidation for this particular cache because it doesn't generally change for users during a process lifetime since everything in it is part of the .NET SDK. Timestamp checks can be surprisingly expensive.

We can add them, but I'd prefer not to if we can get away with opting into "unstable targeting pack mode" in limited circumstances like "the runtime build".

**Comment by ericstj** [comment]
One final thought here.  Runtime might want to benefit from caching too.  Perhaps if you provided some way for us to control the cache key or change something about it so that we could differentiate this scenario.  Could be a way to have cake and eat it too.

**Comment by marcpopMSFT** [comment]
Old issue triage: @ericstj @rainersigwald I assume runtime worked around this a long time ago. Can we close or move to the backlog?

**Comment by ericstj** [comment]
@ViktorHofer what do you think?  I see we still have https://github.com/dotnet/runtime/issues/61451 open, but I know that you've done a lot of refactoring around how we build the shared framework.

**Comment by marcpopMSFT** [comment]
@ViktorHofer feel free to reopen if ya'll still want this.

**Comment by jkoritzinsky** [comment]
It's not the platform manifest but the actual files on disk. Basically, MSBuild would cache the results from disk from the first time we'd build any project that referenced an assembly via the ref pack and all assemblies in the ref pack built later would seemingly never be found.

We are using an environment variable right now to disable the cache, but an MSBuild property would be nicer.
